### PR TITLE
Upgrade to Lucene-9.11.1

### PR DESCRIPTION
**Pull request by iverase**
Lucene 9.11.1 contains important bug fixes we would like to get.

### CODE REVIEW

File: build-tools-internal/version.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 elasticsearch     = 8.15.0
-lucene            = 9.11.0
+lucene            = 9.11.1
```
Comment (benwtrent) [inline build-tools-internal/version.properties:2]: Index version bump?

Comment (iverase) [reply build-tools-internal/version.properties:2]: How I do that these days?

Comment (benwtrent) [reply build-tools-internal/version.properties:2]: `IndexVersions`
```
    public static final IndexVersion UPGRADE_LUCENE_9_11_1 = def(8_510_00_0, Version.LUCENE_9_11_1);
```

**Review by martijnvg — APPROVED** [review]
LGTM

**Review by dnhatn — APPROVED** [review]
LGTM.

**Comment by iverase** [comment]
Thank you!

**Comment by ChrisHegarty** [comment]
Belated LGTM

**Pull request merged**